test/integration/plugin_test.go

```diff
@@ -214,3 +217,157 @@ func TestPluginEssentials(t *testing.T) {
 		return expect404WithNoRoute(t, proxyURL.String(), resp)
 	}, ingressWait, waitTick)
 }
+
+func TestPluginOrdering(t *testing.T) {
+	if !util.GetKongVersion().GTE(semver.MustParse("3.0.0")) || kongEnterpriseEnabled == "" {
+		t.Skip("Kong version does not support plugin ordering")
+	}
+	t.Parallel()
+	ns, cleaner := setup(t)
+	defer func() { assert.NoError(t, cleaner.Cleanup(ctx)) }()
+
+	t.Log("deploying a minimal HTTP container deployment to test Ingress routes")
+	container := generators.NewContainer("httpbin", test.HTTPBinImage, 80)
+	deployment := generators.NewDeploymentForContainer(container)
+	deployment, err := env.Cluster().Client().AppsV1().Deployments(ns.Name).Create(ctx, deployment, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(deployment)
+
+	t.Logf("exposing deployment %s via service", deployment.Name)
+	service := generators.NewServiceForDeployment(deployment, corev1.ServiceTypeLoadBalancer)
+	_, err = env.Cluster().Client().CoreV1().Services(ns.Name).Create(ctx, service, metav1.CreateOptions{})
+	require.NoError(t, err)
+	cleaner.Add(service)
+
+	t.Logf("creating an ingress for service %s with ingress.class %s", service.Name, ingressClass)
+	ingress := generators.NewIngressForService("/httpbin", map[string]string{
+		annotations.IngressClassKey: ingressClass,
+		"konghq.com/strip-path":     "true",
+	}, service)
+	require.NoError(t, clusters.DeployIngress(ctx, env.Cluster(), ns.Name, ingress))
+	cleaner.Add(ingress)
+
+	t.Log("waiting for routes from Ingress to be operational")
+	assert.Eventually(t, func() bool {
+		resp, err := httpc.Get(fmt.Sprintf("%s/httpbin", proxyURL))
+		if err != nil {
+			t.Logf("WARNING: error while waiting for %s: %v", proxyURL, err)
+			return false
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusOK {
+			// now that the ingress backend is routable, make sure the contents we're getting back are what we expect
+			// Expected: "<title>httpbin.org</title>"
+			b := new(bytes.Buffer)
+			n, err := b.ReadFrom(resp.Body)
+			require.NoError(t, err)
+			require.True(t, n > 0)
+			return strings.Contains(b.String(), "<title>httpbin.org</title>")
```

Comment:
I had to have made this comment half unconscious. This comment doesn't belong to this line but to lines actually using `assert`/`require` helpers.

My main concern with those are situations like these:

```go
n, err := b.ReadFrom(resp.Body)
require.NoError(t, err)
```

Is this an error that should warrant the whole test to fail? It might be a consequence of lack of free memory on a box that this test was running on. Should that fail the test? It's not production code: that's test code that does **additional** actions which wouldn't occur in production environments.

> If you're not persuaded, I would ask that we at least not block this PR for this purpose since this is not the only place this is being done, and instead discuss this further in a forum that wont hold up this work (github discussions, slack, zoom, e.t.c.) and if we come to a consensus that we do want to remove these then remove them holistically.

No intention to block this PR on this matter. 👍 I'll bring this topic up in a separate discussion

Reply:
Alright I don't want to overload this one PR I'll follow up in #2629  :+1: